presto-main/src/main/java/com/facebook/presto/security/FileBasedSystemAccessControl.java

```diff
@@ -0,0 +1,323 @@
+import static com.facebook.presto.spi.security.AccessDeniedException.denyDeleteTable;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyDropSchema;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyDropTable;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyDropView;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyGrantTablePrivilege;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyInsertTable;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyRenameColumn;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyRenameSchema;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyRenameTable;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyRevokeTablePrivilege;
+import static com.facebook.presto.spi.security.AccessDeniedException.denySelectTable;
+import static com.facebook.presto.spi.security.AccessDeniedException.denySelectView;
+import static com.facebook.presto.spi.security.AccessDeniedException.denySetCatalogSessionProperty;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyShowSchemas;
+import static com.facebook.presto.spi.security.AccessDeniedException.denyShowTables;
+import static com.google.common.base.Preconditions.checkState;
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static java.util.Objects.requireNonNull;
+
+public class FileBasedSystemAccessControl
+        implements SystemAccessControl
+{
+    public static final String NAME = "file";
+
+    private final List<CatalogAccessControlRule> catalogRules;
+
+    private FileBasedSystemAccessControl(List<CatalogAccessControlRule> catalogRules)
+    {
+        this.catalogRules = catalogRules;
+    }
+
+    public static class Factory
+            implements SystemAccessControlFactory
+    {
+        private static final String CONFIG_FILE_NAME = "security.config-file";
+
+        @Override
+        public String getName()
+        {
+            return NAME;
+        }
+
+        @Override
+        public SystemAccessControl create(Map<String, String> config)
+        {
+            requireNonNull(config, "config is null");
+
+            String configFileName = config.get(CONFIG_FILE_NAME);
+            checkState(
+                    configFileName != null,
+                    "Configuration option %s not found while loading file based system access control", CONFIG_FILE_NAME);
+
+            try {
+                Path path = Paths.get(configFileName);
+                if (!path.isAbsolute()) {
+                    path = path.toAbsolutePath();
+                }
+                return new FileBasedSystemAccessControl(
+                        jsonCodec(FileBasedSystemAccessControlRules.class).fromJson(Files.readAllBytes(path)).getCatalogRules());
```

Comment:
I still would recommend injecting the json codec. they're expensive so you don't want to make multiples of them.